clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -223,6 +249,7 @@ class GCEUtil {
       it.name in forwardingRuleNames
     }
 
+    // Note: this assumes the server group is not associated with any L7 LBs.
     if (foundForwardingRules.size == forwardingRuleNames.size) {
```

Comment:
This assumption is no longer correct now that L7s are introduced. I propose we remove this check and use this call instead of the hideous `queryAllForwardingRulesAndReturnNetwork` thing above it.